js/background.js

```diff
@@ -25,37 +69,33 @@ function _makeRequest(obj) {
     } else {
         obj.headers = addtlHeaders;
     }
-    return fetch(
-        obj.url, 
-        {
-            credentials: 'include',
-            method: obj.method || "GET",
-            headers: {
-                ...obj.headers
-            },
-            body: obj.body
-        }
-    ).then((response) => {
-        if (response.status >= 200 && response.status < 300) {
-            return response.json();
-        }
-        else {
-            throw new Error(response.statusText);
+    return _xhr(obj).then((response) => {
+        if (response.__status__ >= 200 && response.__status__ < 300) {
+
+            return new Promise((resolve, reject) => {
+                setTimeout(() => {
+                    resolve(response);
+                }, 100);
```

Comment:
Adding some randomness to the process.
```suggestion
                }, Math.min(100, Math.round(Math.random() * 100)));
```